doc/api/async_context.md

```diff
@@ -147,6 +147,49 @@ this time will print the stack trace and exit. See
 Creating an async resource within the `onPropagate` callback will result in
 a recursive call to `onPropagate`.
 
+### `asyncLocalStorage.bind(fn [, thisArg])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function} The function to bind to the current execution context.
+* `thisArg` {any}
+
+Binds the given function to the current execution context.
+
+The returned function will have an `asyncResource` property referencing
+the `AsyncResource` to which the function is bound.
```

Comment:
Is this intentional? Shouldn't we distance ourselves from the legacy `AsyncResource`?

Reply:
This is a carryover acknowledging the current state since this defers to AsyncResource under the covers. The next step is going to be deprecating this so we can remove it safely.

Reply:
Since `AsyncLocalStorage.bind` is a new API, I'd find this behavior can be intentionally not documented so that it can be removed without any notice -- although underlying implementation behaves the same with AsyncResource.